### ISSUE TITLE
Do not try to remove sub-package if it was inherited from rings

### ISSUE THREAD

**Issue by nilxam** [openSUSE/osc-plugin-factory#876]
libunbound-devel-mini is linked to unbound and it was inherited from Rings therefore it can not be remove, seems ``osc staging unselect`` has try to remove sub-package(why?) and it failed to work in this case.

```
mlin@rana:~/my_staging/osc-plugin-factory> osc -d staging unselect unbound  
GET https://api.opensuse.org/source/openSUSE:Factory:Staging/_attribute/openSUSE:LockedBy
POST https://api.opensuse.org/source/openSUSE:Factory:Staging/_attribute/openSUSE:LockedBy
GET https://api.opensuse.org/source/openSUSE:Factory:Staging/_attribute/openSUSE:LockedBy
CACHE_GET https://api.opensuse.org/statistics/latest_updated?limit=5000
CACHE_GET https://api.opensuse.org/source/openSUSE:Factory:Staging/dashboard/config?expand=1
CACHE_GET https://api.opensuse.org/source/openSUSE:Factory:Staging/dashboard/ignored_requests?expand=1
CACHE_MISS https://api.opensuse.org/search/project/id?match=starts-with(@name,'openSUSE:Factory:Staging:') (expired)
GET https://api.opensuse.org/search/project/id?match=starts-with(@name,'openSUSE:Factory:Staging:')
CACHE_PUT https://api.opensuse.org/search/project/id?match=starts-with(@name,'openSUSE:Factory:Staging:') openSUSE:Factory:Staging
CACHE_GET https://api.opensuse.org/source/openSUSE:Factory:Staging:A/_meta
CACHE_GET https://api.opensuse.org/source/openSUSE:Factory:Staging:A:DVD/_meta
CACHE_MISS https://api.opensuse.org/source/openSUSE:Factory:Staging:B/_meta (does not exist)
GET https://api.opensuse.org/source/openSUSE:Factory:Staging:B/_meta
CACHE_PUT https://api.opensuse.org/source/openSUSE:Factory:Staging:B/_meta openSUSE:Factory:Staging:B
Unselecting "492010" from "openSUSE:Factory:Staging:B"
CACHE_GET https://api.opensuse.org/source/openSUSE:Factory:Staging:B/_meta
CACHE_DELETE_PROJECT https://api.opensuse.org openSUSE:Factory:Staging:B
PUT https://api.opensuse.org/source/openSUSE:Factory:Staging:B/_meta?force=1
CACHE_GET https://api.opensuse.org/source/openSUSE:Factory:Rings:0-Bootstrap?nofilename=1&view=info
CACHE_GET https://api.opensuse.org/source/openSUSE:Factory:Rings:1-MinimalX?nofilename=1&view=info
CACHE_MISS https://api.opensuse.org/source/openSUSE:Factory:Rings:2-TestDVD?nofilename=1&view=info (does not exist)
GET https://api.opensuse.org/source/openSUSE:Factory:Rings:2-TestDVD?nofilename=1&view=info
CACHE_PUT https://api.opensuse.org/source/openSUSE:Factory:Rings:2-TestDVD?nofilename=1&view=info openSUSE:Factory:Rings:2-TestDVD
CACHE_MISS https://api.opensuse.org/source/openSUSE:Factory:Staging:B:DVD/_meta (does not exist)
GET https://api.opensuse.org/source/openSUSE:Factory:Staging:B:DVD/_meta
CACHE_PUT https://api.opensuse.org/source/openSUSE:Factory:Staging:B:DVD/_meta openSUSE:Factory:Staging:B:DVD
CACHE_DELETE_PROJECT https://api.opensuse.org openSUSE:Factory:Staging:B:DVD
DELETE https://api.opensuse.org/source/openSUSE:Factory:Staging:B:DVD/unbound?comment=Removing+from+openSUSE%3AFactory%3AStaging%3AB%2C+re-evaluation+needed&force=1
CACHE_DELETE_PROJECT https://api.opensuse.org openSUSE:Factory:Rings:2-TestDVD
POST https://api.opensuse.org/source/openSUSE:Factory:Rings:2-TestDVD/unbound?cmd=showlinked
DELETE https://api.opensuse.org/source/openSUSE:Factory:Staging:B/libunbound-devel-mini?comment=Removing+from+openSUSE%3AFactory%3AStaging%3AB%2C+re-evaluation+needed&force=1
GET https://api.opensuse.org/source/openSUSE:Factory:Staging/_attribute/openSUSE:LockedBy
POST https://api.opensuse.org/source/openSUSE:Factory:Staging/_attribute/openSUSE:LockedBy
Server returned an error: HTTP Error 404: Not Found
Date: Wed, 03 May 2017 08:37:35 GMT
Server: Apache/2.4.23 (Linux/SUSE)
Strict-Transport-Security: max-age=31536000
Cache-Control: no-cache
Vary: Accept-Encoding
X-XSS-Protection: 1; mode=block
X-Opensuse-Runtimes: {"view":1.4054668135941029,"db":17.118597,"backend":0,"xml":0}
X-Request-Id: a4c4136c-5828-475b-b5f3-afc618fd4057
X-Opensuse-APIVersion: 2.8.51.git20170419.897b9e1
X-Runtime: 0.054827
X-Frame-Options: SAMEORIGIN
X-Content-Type-Options: nosniff
X-Opensuse-Errorcode: unknown_package
X-Powered-By: Phusion Passenger 5.0.30
Status: 404 Not Found
Content-Type: application/xml; charset=utf-8
Connection: close
Transfer-Encoding: chunked

<status code="unknown_package">
  <summary>openSUSE:Factory:Staging:B/libunbound-devel-mini</summary>
</status>

openSUSE:Factory:Staging:B/libunbound-devel-mini
```

**Comment by jberry-suse** [comment]
From [`StagingAPI.rm_from_prj()` line 764](https://github.com/openSUSE/osc-plugin-factory/blob/22ec112225255700b58b63f701da53b8d8df5a00/osclib/stagingapi.py#L764):

```python
for sub_prj, sub_pkg in self.get_sub_packages(package):
    sub_prj = self.map_ring_package_to_subject(project, sub_pkg)
    if sub_prj != subprj:  # if different to the main package's prj
        delete_package(self.apiurl, sub_prj, sub_pkg, force=True, msg=msg)
```

This explains it assuming `get_sub_packages()` returned `libunbound-devel-mini`. I ran the same command it would have issued (and based on log did) and see the following.

```
$ osc api -X POST /source/openSUSE:Factory:Rings:2-TestDVD/unbound?cmd=showlinked
<collection>
  <package project="openSUSE:Factory:Rings:1-MinimalX" name="libunbound-devel-mini"/>
</collection>
```

The condition guarding the delete was then `True` since `sub_prj != subprj`. The enlightening variable names do not help. :)

In this case one is in `:DVD` and the other is in the main letter project so the sub projects do not match and it attempts to delete them. I am not exactly sure how this condition makes sense as it seems more important to know if the subpackages were created from the staged package rather than if they are in a different subproject.

I would have to dig into `select` to determine what the current behavior there is for subpackages.

What is the desired outcome here? Perhaps I am missing something, but it would seem both `unbound` and its subpackage `libunbound-devel-mini` should have been updated during the `select` and then both `unselect`ed, but based on the title of this issue it sounds like the `select` behavior is desired and `unselect` needs to be changed to detect if subpackages are linked to rings and thus not representing `select`.

I would assume the condition `if sub_prj != subprj` should be entirely disregarded and replaced with such a link check.
